src/sentry/static/sentry/app/components/contextData.jsx

```diff
@@ -73,6 +72,42 @@ function analyzeStringForRepr(value) {
   return rv;
 }
 
+class ToggleWrap extends React.Component {
+  static propTypes = {
+    highUp: PropTypes.bool,
+    wrapClassName: PropTypes.string,
+  };
+
+  state = {toggled: false};
+
+  render() {
+    if (React.Children.count(this.props.children) === 0) {
+      return null;
+    }
+    if (this.props.highUp) {
+      return this.props.children;
+    }
+
+    const classes = ['val-toggle'];
+    if (this.state.toggled) {
+      classes.push('val-toggle-open');
+    }
+    return (
+      <span className={classes.join(' ')}>
+        <a
+          href="#"
+          className="val-toggle-link"
+          onClick={evt => {
+            this.setState({toggled: !this.state.toggled});
```

Comment:
This is probably fine, but typically when using existing state in a `setState` you'll want to use the callback version

```jsx
this.setState(state => ({toggled: !state.toggled}))
```

[Here's why if you're curious](https://medium.com/@wereHamster/beware-react-setstate-is-asynchronous-ce87ef1a9cf3)

Reply:
Ah, it's been quite a while since I wrote React code. Thanks a lot for catching all this! :D